receiver/k8sclusterreceiver/e2e_test.go

```diff
@@ -108,17 +110,20 @@ func TestE2E(t *testing.T) {
 	)
 }
 
-func waitForData(t *testing.T, entriesNum int, mc *consumertest.MetricsSink) {
+func setUpSync(t *testing.T) (*consumertest.MetricsSink, func()) {
+	sink := new(consumertest.MetricsSink)
 	f := otlpreceiver.NewFactory()
 	cfg := f.CreateDefaultConfig().(*otlpreceiver.Config)
 
-	rcvr, err := f.CreateMetricsReceiver(context.Background(), receivertest.NewNopCreateSettings(), cfg, mc)
+	rcvr, err := f.CreateMetricsReceiver(context.Background(), receivertest.NewNopCreateSettings(), cfg, sink)
 	require.NoError(t, rcvr.Start(context.Background(), componenttest.NewNopHost()))
 	require.NoError(t, err, "failed creating metrics receiver")
-	defer func() {
+	return sink, func() {
 		assert.NoError(t, rcvr.Shutdown(context.Background()))
-	}()
+	}
+}
 
+func waitForData(t *testing.T, entriesNum int, mc *consumertest.MetricsSink) {
```

Comment:
Looks like this is failing most of time now, with or without this change:
https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/workflows/e2e-tests.yml

Should we skip this test for now?

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29916

Reply:
yes, this change is not fixing the issue, it's too bad.